theano/sandbox/cuda/blas.py

```diff
@@ -2163,3 +2163,183 @@ def c_support_code_apply(self, node, nodename):
             }
         }
         """ % locals()
+
+
+class GpuDownsampleFactorMaxGradGrad(GpuOp):
+    """
+    Implement the grad of downsample with max on the gpu.
+    """
+    __props__ = ('ds', 'ignore_border')
+    
+    def __init__(self, ds, ignore_border):
+        self.ds = tuple(ds)
+        self.ignore_border = ignore_border
+
+    def make_node(self, x, z, gx):
+        return Apply(self, [x, z, gx], [x.type()])
```

Comment:
add `as_cudandarray_variable` on `x`, `z` and `gx` or at least an assert that makes sure these are CudaNdarrayVariables